test/rules/no-array-literal-hole/test.ts.lint

```diff
@@ -0,0 +1,5 @@
+[1, , 3];
+   ~ [Array has a missing element.]
+
+// Destructuring allowed.
+const [x, , z] = arr;
```

Comment:
That's only destructuring in declarations (ArrayBindingPattern). Destructuring in expression uses ArrayLiteralExpression, see my other comment above.